autoload/codefmt/autopep8.vim

```diff
@@ -67,29 +67,18 @@ function! codefmt#autopep8#GetFormatter() abort
 
     call maktaba#ensure#IsNumber(a:startline)
     call maktaba#ensure#IsNumber(a:endline)
-    let l:lines = getline(1, line('$'))
 
     if s:autopep8_supports_range
-      let l:cmd = [l:executable, '--range', ''.a:startline, ''.a:endline, '-']
-      let l:input = join(l:lines, "\n")
+      call codefmt#formatterhelpers#Format(maktaba#syscall#Create([
+          \ l:executable,
+          \ '--range', string(a:startline), string(a:endline),
+          \ '-']))
     else
-      let l:cmd = [l:executable, '-']
-      " Hack range formatting by formatting range individually, ignoring context.
-      let l:input = join(l:lines[a:startline - 1 : a:endline - 1], "\n")
+      call codefmt#formatterhelpers#AttemptFakeRangeFormatting(
+          \ a:startline,
+          \ a:endline,
+          \ maktaba#syscall#Create([l:executable, '-']))
```

Comment:
Any reason not to prefer just `[l:executable, '-']` here, and similarly `l:cmd` (etc) elsewhere, as you have for buildifier?

Reply:
Yep, fair question. It'd save keystrokes but lose flexibility we'll want. For example, it wouldn't accommodate the `WithCwd` in zprint that I just rebased in.

Reply:
I gotcha now. Updated.